code/datums/components/style/style_meter.dm

```diff
@@ -8,14 +8,20 @@
 	icon = 'icons/obj/clothing/glasses.dmi'
 	/// The style meter component we give.
 	var/datum/component/style/style_meter
+	/// Mutable appearance added to the attached glasses
+	var/mutable_appearance/meter_appearance
+
+/obj/item/style_meter/Initialize(mapload)
+	. = ..()
+	meter_appearance = mutable_appearance(icon, icon_state)
 
 /obj/item/style_meter/afterattack(atom/movable/attacked_atom, mob/user, proximity_flag, click_parameters)
 	. = ..()
 	if(!istype(attacked_atom, /obj/item/clothing/glasses))
 		return
 
 	forceMove(attacked_atom)
-	attacked_atom.vis_contents += src
```

Comment:
why

Reply:
It didn't work at all, the meter didn't appear
![image](https://user-images.githubusercontent.com/41448081/231615008-d5b1d927-8083-4f16-b5ac-63a48687fdc8.png)

new method works though